source/MRViewer/MRPlaneWidget.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+#include "MRMesh/MRBox.h"
+#include "MRMesh/MRPlane3.h"
+#include "MRViewer/MRViewer.h"
+
+namespace MR
+{
+class PlaneWidget : public MultiListener<MouseDownListener, MouseMoveListener, MouseUpListener>
+{
+    std::shared_ptr<ObjectMesh> planeObj_;
+    Plane3f plane_;
+    Box3f box_;
+    Vector3f cameraUp3Old_;
+    
+    std::function<void( const Plane3f& )> onPlaneUpdate_;
+
+    bool pressed_ = false;
+    ImVec2 startMousePos_;
+    ImVec2 endMousePos_;
+
+public:
+    MRVIEWER_API PlaneWidget( const Plane3f& plane, const Box3f& box, std::function<void( const Plane3f& )> = nullptr );
+    MRVIEWER_API ~PlaneWidget();
```

Comment:
I think it is better to have functions like `create/reset` instead of RAII to support `defined/undefined` logic